### PR TITLE
3.0.0

### LINKED ISSUE #21: Use `ts-bridge` as build tool, set `moduleResolution` to `NodeNext`
## Explanation

As part of our OKR for upgrading TypeScript to v5.0+ (Q2 2024 O3KR4), we are updating our core dependencies to generate builds and type declarations for both CJS and ESM.

## References

- Same update made to `@metamask/utils`: https://github.com/MetaMask/utils/pull/182
- Required to update core repo's `moduleResolution` setting to `NodeNext`:
  - Blocks: https://github.com/MetaMask/core/issues/3651

### PR DESCRIPTION
**Pull request by MajorLift**
Release candidate for v3.0.0.

- Closes #18

### CODE REVIEW

**Review by cryptodev-2s — DISMISSED** [review]
LGTM!

**Review by mcmire — DISMISSED** [review]
LGTM as well.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,9 +7,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
 ### Added
 
-- Newly exports the following types: `AnyStruct`, `EnumSchema`, `InferStructTuple`, `IsExactMatch`, `IsMatch`, `IsRecord`, `IsTuple`, `ObjectSchema`, `OmitBy`, `Optionalize`, `PickBy`, `Simplify`, `UnionToIntersection`.
+- Newly exports the following types: `AnyStruct`, `EnumSchema`, `InferStructTuple`, `IsExactMatch`, `IsMatch`, `IsRecord`, `IsTuple`, `ObjectSchema`, `OmitBy`, `Optionalize`, `PickBy`, `Simplify`, `UnionToIntersection` ([#18](https://github.com/MetaMask/superstruct/pull/18))
+
+### Changed
+
+- **BREAKING:** Expose separate build entry points and type declarations for CommonJS and ESM via package manifest `exports`. ([#18](https://github.com/MetaMask/superstruct/pull/18))
+  - When imported as a module, this package can now resolve as both CommonJS and ESM.
```
Comment (MajorLift) [inline CHANGELOG.md]:
```suggestion
### Fixed

- **BREAKING:** Expose separate build entry points and type declarations for CommonJS and ESM via package manifest `exports` ([#18](https://github.com/MetaMask/superstruct/pull/18)).
```

**Review by cryptodev-2s — APPROVED** [review]
LGTM!

**Pull request merged**